xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/script/RefactoringScriptService.java

```diff
@@ -271,7 +271,7 @@ public Job copy(EntityReference source, EntityReference destination)
      */
     public Job copyAs(CopyRequest request)
     {
-        return execute(RefactoringJobs.COPY_AS, request);
+        return execute(RefactoringJobs.COPY, request);
```

Comment:
If that's the right fix then the method name is very wrong.

Reply:
We don't have a specific job to distinguish between copy and copyAs. So we have to use the same component.

Now this entire concept of "copy" VS "copyAs" seems to add useless complexity, even the doc does not clearly specify the distinction:


> Copy a Space
>```
>#set ($source = $services.model.resolveSpace('Path.To.Source'))
>#set ($destination = $services.model.resolveSpace('Path.To.New.Parent'))
>$services.refactoring.copy($source, $destination).join()
>```
>Copy a Space As
>```
>#set ($source = $services.model.resolveSpace('Path.To.Source'))
>#set ($destination = $services.model.resolveSpace('Path.To.New.Name'))
>$services.refactoring.copyAs($source, $destination).join()
>```

So we might as well simplify the API by pushing to legacy copyAs.

Reply:
> We don't have a specific job to distinguish between copy and copyAs.

So this means that the RefactoringJobs.COPY_AS does not make much sense in the first place since its javadoc is false.

Reply:
> So this means that the RefactoringJobs.COPY_AS does not make much sense in the first place since its javadoc is false.

As it was the case before for RefactoringJobs.COPY :) It was only used for the request, never for the job itself.

Reply:
ping @mflorea WDYT to put the copyAs to legacy since the distinction does not look really meaningful now?

Reply:
> even the doc does not clearly specify the distinction

The distinction is there and it's clear to me:

* for the **copy** operation you need to specify the **parent** (the location where to copy). The page name doesn't change.
* for the **copy as** operation you need to specify the **new reference**. The page name can change.

For me it makes a lot of sense to have both **copy** and **copyAs** at the level of the script service API, even though both use the same (copy) job under the hood, because **the input is different**.

Reply:
OK to follow this distinction, but it wasn't clear at all in the code: copyAs was previously calling a RenameJob without deletion instead of a MoveJob for copy, but besides that the destination was set the same way. 
And right now, after the refactoring a RenameJob *is* a CopyJob without deletion: which might mean my CopyJob is actually already a *CopyAsJob*.

So, in order to simplify, I propose to add a new boolean in CopyRequest:
```
keepPageName()
```

if this boolean is true, the CopyJob will work as a Copy, else it will work as a CopyAs. Then calling the `createCopyRequest` or `createCopyAsRequest` will always create a `CopyRequest` but with the proper boolean set. WDYT?

Reply:
Actually I even don't need it. I just changed the destination in case of a copy in the request factory. Tell me what do you think, I did not test it.